### ISSUE TITLE
Relay transactions and check them for inclusion

### ISSUE THREAD

**Issue by rustaceanrob** [2140-dev/kyoto#220]
To enhance privacy, the node may relay all transactions received by connected nodes. There is an added bonus that incoming transactions to the users `Scripts` could be checked for as transactions are received.

**Comment by rustaceanrob** [comment]
More technical overhead to consider. Transactions are announced with an `inv` followed by a `tx` message. There would almost certainly have to be a separate task that would listen for transaction relay and do this back-and-forth.

**Comment by nyonson** [comment]
Would this significantly up the amount of bandwidth kyoto uses? It isn't hangin on to the txs (mempool-esque), but it will still be a lot more chatty right?

**Comment by rustaceanrob** [comment]
Yeah, because all transactions are first advertised by an `inv`, there would be one message exchange to download the transaction data, and, if deemed relevant due to the outputs, there would be `N` message exchanges to download the transactions attributed to the inputs (`N` is the number of inputs). Since the number of inputs can be quite large, especially for exchanges, this is definitely an opt-in feature and a longer term goal. Not really on my radar at the moment.

**Comment by rustaceanrob** [comment]
Future versions will be script-less within the client. Closing, although an interesting side-car application may still be possible.
